google/cloud/aiplatform/metadata/metadata.py

```diff
@@ -30,13 +30,16 @@ class _MetadataService:
     def __init__(self):
         self._experiment = None
         self._run = None
+        self._metric = None
 
     def set_experiment(self, experiment: str):
         _MetadataStore.get_or_create()
         context = _Context.get_or_create(
             resource_id=experiment,
+            display_name=experiment,
             schema_title=constants.SYSTEM_EXPERIMENT,
             schema_version=constants.SCHEMA_VERSIONS[constants.SYSTEM_EXPERIMENT],
+            metadata={constants.UI_DETECTION_KEY: constants.UI_DETECTION_VALUE},
         )
         self._experiment = context.name
```

Comment:
If we went with instance methods, we could reference the context directly and not have to indirectly track the name.

Reply:
I added an instance field to the class to track the experiment context. I think we would still want to keep the string/ID name here as those names are used to generate run/metric resource names.

Reply:
I agree with Sasha, I think we should either track resources, or track names... no need to track both name and resources...

Reply:
Ok, I updated all 3 to store the whole resource instead of the name. And also added a comments in the log_xxx method to explain why we want to query a new resource instead of using the one we stored.